### PR TITLE
[BENCH t01 deepseek-v3.2-c] feat(tags): Add common_tags() helper for standardized resource tagging

### LINKED ISSUE #34: [BENCH t01 deepseek-v3.2-c] feat(tags): Add common_tags() helper for standardized resource tagging
### Objective

feat(tags): Add common_tags() helper for standardized resource tagging

### Acceptance criteria

- [ ] Implementation matches all behaviors described in the task body (see Notes)
- [ ] All named tests pass the verification command
- [ ] No dependencies added unless absolutely required

### Out-of-scope / non-goals

- Do NOT modify files beyond the expected set below
- Do NOT add third-party dependencies unless required by the task body
- Do NOT refactor unrelated code in the touched files

### Files expected to change

- `infiquetra_aws_infra/tagging.py`
- `tests/test_tagging.py`

### Tests to add or update

- See the task body for the named test cases

### Verification

```bash
cd ~/workspace/infiquetra/infiquetra-aws-infra
# Language-appropriate test command (flutter test / pytest / etc.)
```

## Task

Add `common_tags(env: str, team: str, project: str) -> dict[str, str]` to `infiquetra_aws_infra/tagging.py`.

## Requirements

- Return a dict with the keys: `Environment`, `Team`, `Project`, `ManagedBy`.
- `ManagedBy` must always be `"CDK"`.
- `env` must be one of `"dev"`, `"staging"`, `"prod"` — raise `ValueError` on others.
- Strip/normalize trailing whitespace on all inputs.
- No external dependencies.

## Examples

- `common_tags("dev", "platform", "auth")` → `{"Environment": "dev", "Team": "platform", "Project": "auth", "ManagedBy": "CDK"}`
- `common_tags("test", "x", "y")` → raises `ValueError`
- `common_tags("dev  ", " platform ", "auth")` → normalized keys

## Tests

Add `tests/test_tagging.py` with pytest: valid dev/staging/prod, ValueError for bad env, whitespace normalization, all keys present.

Scope: small (<40 lines).

### Context library links

- Benchmark task id: t01

### PR DESCRIPTION
**Pull request by namredips**
## Linked card

Closes #34

## What changed

- Added `infiquetra_aws_infra/tagging.py` with `common_tags()` helper.
- Added `tests/test_tagging.py` with full coverage for `common_tags()`.
- Implemented environment validation (dev, staging, prod) and whitespace normalization.
- Ensured `ManagedBy: CDK` is always present.

## How verified

Ran `pytest` with the following output:
```
....                                                                     [100%]
4 passed in 0.01s
```

Verified linting with `ruff check` and typing with `mypy`.

## Acceptance criteria coverage

- AC 1: Implementation matches all behaviors described in the task body: ✓ addressed in `infiquetra_aws_infra/tagging.py`
- AC 2: All named tests pass the verification command: ✓ addressed in `tests/test_tagging.py`
- AC 3: No dependencies added unless absolutely required: ✓ no new dependencies added

## Non-goals acknowledged

- Do NOT modify files beyond the expected set: not violated
- Do NOT add third-party dependencies: not violated
- Do NOT refactor unrelated code: not violated

### Coverage

- AC 1: Implementation matches all behaviors described in the task body (see Notes): ✓ addressed in `infiquetra_aws_infra/tagging.py`
- AC 2: All named tests pass the verification command: ✓ addressed in `tests/test_tagging.py`
- AC 3: No dependencies added unless absolutely required: ✓ addressed (no changes to pyproject.toml)

### CODE REVIEW

**Comment by namredips** [comment]
## ✅ PR review round 1 — MERGE-READY

**Overall:** 8.68/10 · **HEAD:** `430f4d06`

_Per-reviewer scores: ac_verification 0.0 · code_quality 8.8 · plan_fidelity 7.6 · robustness 8.8 · security_code 8.4 · python_types 9.0 · test_presence 9.5_

### Reviewer verdicts

<details open>
<summary>🔄 <strong>ac_verification</strong> — 0.00/10</summary>

[reviewer LLM failed after retries — verdict is synthetic, excluded from aggregate. Error: Command '['/Users/jefcox/.local/bin/hermes', 'chat', '-q', '[SYSTEM: You are the `ac_verification` reviewer, evaluating an **ACTUAL PULL\nREQUEST** — code that exists. The agent has implemented the approved\nplan and pushed commits. Your job is to judge whether the code is\nmerge-ready from your rubric\'s perspective.\n\nYou receive the card as structured fields:\n\n**Objective:** feat(tags): Add common_tags() helper for standardized resource tagging\n\n**Acceptance criteria:**\n- [ ] Implementation matches all behaviors described in the task body (see Notes)\n- [ ] All named tests pass the verification command\n- [ ] No dependencies added unless absolutely required\n\n**Out-of-scope / non-goals:**\n- Do NOT modify files beyond the expected set below\n- Do NOT add third-party dependencies unless required by the task body\n- Do NOT refactor unrelated code in the touched files\n\n**Files expected to change:**\n- `infiquetra_aws_infra/tagging.py`\n- `tests/test_tagging.py`\n\n**Tests to add/update:**\n- See the task body for the named test cases\n\n**Verification:**\n```bash\ncd ~/workspace/infiquetra/infiquetra-aws-infra\n# Language-appropriate test command (flutter test / pytest / etc.)\n```\n\n**Notes / conventions:**\n## Task\n\nAdd `common_tags(env: str, team: str, project: str) -> dict[str, str]` to `infiquetra_aws_infra/tagging.py`.\n\n## Requirements\n\n- Return a dict with the keys: `Environment`, `Team`, `Project`, `ManagedBy`.\n- `ManagedBy` must always be `"CDK"`.\n- `env` must be one of `"dev"`, `"staging"`, `"prod"` — raise `ValueError` on others.\n- Strip/normalize trailing whitespace on all inputs.\n- No external dependencies.\n\n## Examples\n\n- `common_tags("dev", "platform", "auth")` → `{"Environment": "dev", "Team": "platform", "Project": "auth", "ManagedBy": "CDK"}`\n- `common_tags("test", "x", "y")` → raises `ValueError`\n- `common_tags("dev  ", " platform ", "auth")` → normalized keys\n\n## Tests\n\nAdd `tests/test_tagging.py` with pytest: valid dev/staging/prod, ValueError for bad env, whitespace normalization, all keys present.\n\nScope: small (<40 lines).\n\n**Context library links:** - Benchmark task id: t01\n\n\n\n\n\n\nAnd the **approved plan** the agent was supposed to implement:\n\n## Files to modify\n\n- `infiquetra_aws_infra/tagging.py`\n  - Verified with `gh api repos/infiquetra/infiquetra-aws-infra/contents/infiquetra_aws_infra`: the package directory exists and currently contains `__init__.py`, `organization_stack.py`, and `sso_stack.py`; `tagging.py` does not exist yet.\n  - Create this new module to define `common_tags(env: str, team: str, project: str) -> dict[str, str]`.\n\n- `tests/test_tagging.py`\n  - Verified with `gh api repos/infiquetra/infiquetra-aws-infra/contents`: the repository root exists but does not currently include a top-level `tests/` directory.\n  - Verified with `gh api repos/infiquetra/infiquetra-aws-infra/contents/tests`: GitHub returned 404, so `tests/test_tagging.py` is a new file and its parent `tests/` directory must be created.\n  - Create this pytest file for the named `common_tags()` behaviors.\n\n## Approach\n\n1. Add `infiquetra_aws_infra/tagging.py` as a small standalone helper module with a single public function, `common_tags(env: str, team: str, project: str) -> dict[str, str]`.\n   - Normalize each incoming string with `strip()` before any validation or return-value construction.\n   - Store the normalized values in local variables such as `normalized_env`, `normalized_team`, and `normalized_project` so the validation path and returned dict use the same cleaned inputs.\n   - Keep the module dependency-free; no new imports should be required beyond what Python already provides.\n\n2. In `common_tags()`, validate the normalized environment value against the exact allowed set `{"dev", "staging", "prod"}`.\n   - Only if the normalized environment is one of those three values, proceed to build and return the tags dict.\n   - If the normalized environment is anything else, the function must raise `ValueError`; it must not silently coerce, default, or return partial data.\n\n3. Return a dict literal from `common_tags()` with exactly these keys and values:\n   - `"Environment": normalized_env`\n   - `"Team": normalized_team`\n   - `"Project": normalized_project`\n   - `"ManagedBy": "CDK"`\n   This keeps the implementation aligned with the examples and avoids unrelated refactoring in the existing stack files.\n\n4. Add `tests/test_tagging.py` under a new top-level `tests/` directory, following the repository’s pytest configuration already present in `pyproject.toml` (`testpaths = ["tests"]`, `python_files = ["test_*.py", "*_test.py"]`).\n   - Add one parametrized or equivalent test covering valid environments `dev`, `staging`, and `prod`.\n   - Add one test asserting `ValueError` for an invalid environment such as `"test"`.\n   - Add one test asserting whitespace normalization, using the example shape `common_tags("dev  ", " platform ", "auth")`.\n   - Add one test asserting all required keys are present and that `"ManagedBy"` is always `"CDK"`.\n\n5. Keep the change set scoped to the two expected files only.\n   - Do not edit `pyproject.toml`, stack files, or package exports unless the task becomes impossible without it.\n   - If creating `tests/test_tagging.py` under a new `tests/` directory fails for filesystem or path reasons, the task must fail rather than relocating the test file elsewhere, because `pyproject.toml` is already configured to collect from `tests/`.\n\n6. Run verification from the repository root after both files are in place.\n   - Command: `cd ~/workspace/infiquetra/infiquetra-aws-infra && pytest`\n   - Only if `pytest` exits successfully should the task be considered complete.\n   - If `pytest` fails, errors during collection, or reports any failing test, the task must fail and be fixed before completion.\n\n## Acceptance criteria coverage\n\n- AC 1: Implementation matches all behaviors described in the task body\n  - Covered by steps 1-3.\n  - Mechanisms:\n    - `common_tags(env: str, team: str, project: str) -> dict[str, str]` is added in `infiquetra_aws_infra/tagging.py`.\n    - `normalized_env = env.strip()`, `normalized_team = team.strip()`, and `normalized_project = project.strip()` satisfy the whitespace normalization requirement.\n    - `if normalized_env not in {"dev", "staging", "prod"}: raise ValueError(...)` satisfies the invalid-environment requirement.\n    - The returned dict literal with keys `Environment`, `Team`, `Project`, and `ManagedBy`, with `"ManagedBy": "CDK"`, satisfies the required output shape.\n\n- AC 2: All named tests pass the verification command\n  - Covered by steps 4 and 6.\n  - Mechanisms:\n    - `tests/test_tagging.py` will contain explicit pytest coverage for valid dev/staging/prod inputs, invalid env raising `ValueError`, whitespace normalization, and all keys present.\n    - `pyproject.toml` already declares `testpaths = ["tests"]`, so `pytest` from the repo root will collect the new file at `tests/test_tagging.py`.\n    - The verification command `cd ~/workspace/infiquetra/infiquetra-aws-infra && pytest` is the pass/fail gate.\n\n- AC 3: No dependencies added unless absolutely required\n  - Covered by steps 1 and 5.\n  - Mechanisms:\n    - `common_tags()` uses only built-in string operations (`strip()`) and a dict literal.\n    - No dependency files are touched, and no imports beyond standard Python behavior are needed.\n    - Tests rely on existing pytest support already declared in `pyproject.toml`.\n\n## Out-of-scope acknowledged\n\n- Do NOT modify files beyond the expected set below — this plan changes only `infiquetra_aws_infra/tagging.py` and `tests/test_tagging.py`.\n- Do NOT add third-party dependencies unless required by the task body — this plan adds no dependencies.\n- Do NOT refactor unrelated code in the touched files — this plan adds the helper and focused tests only.\n\n## Tests / verification\n\nNew test file:\n- `tests/test_tagging.py`\n\nPlanned test functions:\n- `test_common_tags_returns_expected_values_for_valid_envs`\n- `test_common_tags_raises_value_error_for_invalid_env`\n- `test_common_tags_strips_whitespace_from_inputs`\n- `test_common_tags_includes_all_required_keys`\n\nVerification commands:\n- `cd ~/workspace/infiquetra/infiquetra-aws-infra`\n- `pytest`\n\nAnd the **PR diff** the agent produced:\n\n```diff\ndiff --git a/infiquetra_aws_infra/tagging.py b/infiquetra_aws_infra/tagging.py\nnew file mode 100644\nindex 0000000..25843a9\n--- /dev/null\n+++ b/infiquetra_aws_infra/tagging.py\n@@ -0,0 +1,30 @@\n+def common_tags(env: str, team: str, project: str) -> dict[str, str]:\n+    """\n+    Generate common tags for AWS resources.\n+\n+    Args:\n+        env: The environment (dev, staging, prod).\n+        team: The team responsible for the resource.\n+        project: The project name.\n+\n+    Returns:\n+        A dictionary of tags.\n+\n+    Raises:\n+        ValueError: If the environment is not one of dev, staging, or prod.\n+    """\n+    normalized_env = env.strip()\n+    normalized_team = team.strip()\n+    normalized_project = project.strip()\n+\n+    valid_envs = {"dev", "staging", "prod"}\n+    if normalized_env not in valid_envs:\n+        msg = f"Invalid environment \'{normalized_env}\'. Must be one of {valid_envs}"\n+        raise ValueError(msg)\n+\n+    return {\n+        "Environment": normalized_env,\n+        "Team": normalized_team,\n+        "Project": normalized_project,\n+        "ManagedBy": "CDK",\n+    }\ndiff --git a/tests/test_tagging.py b/tests/test_tagging.py\nnew file mode 100644\nindex 0000000..7aca084\n--- /dev/null\n+++ b/tests/test_tagging.py\n@@ -0,0 +1,28 @@\n+import pytest  # type: ignore\n+\n+from infiquetra_aws_infra.tagging import common_tags\n+\n+\n+def test_common_tags_returns_expected_values_for_valid_envs() -> None:\n+    for env in ["dev", "staging", "prod"]:\n+        tags = common_tags(env, "platform", "auth")\n+        assert tags["Environment"] == env\n+        assert tags["Team"] == "platform"\n+        assert tags["Project"] == "auth"\n+        assert tags["ManagedBy"] == "CDK"\n+\n+def test_common_tags_raises_value_error_for_invalid_env() -> None:\n+    with pytest.raises(ValueError, match="Invalid environment \'test\'"):\n+        common_tags("test", "x", "y")\n+\n+def test_common_tags_strips_whitespace_from_inputs() -> None:\n+    tags = common_tags("dev  ", " platform ", "auth")\n+    assert tags["Environment"] == "dev"\n+    assert tags["Team"] == "platform"\n+    assert tags["Project"] == "auth"\n+\n+def test_common_tags_includes_all_required_keys() -> None:\n+    tags = common_tags("prod", "infra", "networking")\n+    expected_keys = {"Environment", "Team", "Project", "ManagedBy"}\n+    assert set(tags.keys()) == expected_keys\n+    assert tags["ManagedBy"] == "CDK"\n\n```\n\n## How to evaluate\n\n**Step 1 — Mechanical AC coverage check.**\nFor each acceptance criterion listed in the card, classify the diff\'s\ntreatment:\n- ADDRESSED — the diff\'s code produces the observable behavior this AC\n  demands (and a test in the diff proves it, or the card said "no\n  tests required" with a rationale you can verify).\n- UNADDRESSED — the diff doesn\'t produce the behavior; agent missed\n  or skipped this AC.\n- PARTIALLY — the diff covers the AC for the common case but a named\n  edge case isn\'t handled.\n\n**Step 2 — Non-goals check.**\nVerify the diff does NOT do any work on any non-goal. Scope creep\n(diff modifies files the card explicitly said to avoid, adds features\nthe non-goals forbid) is grounds for REVISE or BLOCK.\n\n**Step 3 — Rubric-specific qualitative review.**\nApply your rubric below as a lens over the diff. Your rubric is the\nauthoritative guide to what YOU care about. Other reviewers cover\nother lenses — stay in your lane.\n\nYour rubric:\n\n---\nphases: [pr]\napplicability: always\n---\n# AC Verification lens (PR-phase)\n\nYour focus: **for each Acceptance Criterion in the card, does the\ndiff (+ its tests) produce the observable behavior the AC demands?**\n\nYou are the customer-facing reviewer. ACs are the contract between\nthe card author and the automation. Plan approval didn\'t matter if\nthe diff doesn\'t actually satisfy the ACs.\n\n## How this differs from `plan_fidelity`\n\n- `plan_fidelity`: did the agent implement what the plan said?\n- **You (`ac_verification`)**: does the diff satisfy what the card\'s\n  ACs demand?\n\nA diff can be plan-faithful but AC-incomplete (the plan had a gap\nthe reviewers missed). Conversely, a diff can deviate from the plan\nbut still satisfy every AC (agent discovered a better path). You\njudge AC outcomes, not plan-step completion.\n\n## How to evaluate\n\nFor **each AC** in the card (you have them in your system prompt),\ndo the following:\n\n1. **Identify the observable.** What would a human verifier look for\n   to confirm this AC holds? An exit code? A specific output? A\n   behavior under a named input?\n2. **Find the evidence in the diff.** Does the code contain the\n   behavior? Does a test in the diff prove the behavior?\n3. **Check the Verification command.** The card has an exact\n   command in its Verification section. Does the diff make that\n   command pass?\n4. **Record coverage.** Return `ac_coverage` in your JSON verdict:\n   each AC number → ADDRESSED | PARTIALLY | UNADDRESSED.\n\n## What counts as "addressed"\n\n- **ADDRESSED**: the behavior the AC demands is visible in the\n  diff\'s code, AND a test in the diff proves it (OR the card said\n  "no new tests required" with a rationale you can verify holds).\n- **PARTIALLY**: the behavior is present but a named edge case the\n  AC mentions isn\'t handled (e.g., "handles malformed input" when\n  only well-formed input is in the code path).\n- **UNADDRESSED**: the AC\'s observable behavior is not in the diff,\n  or there\'s no way to verify it from what\'s here.\n\n## What to look for\n\n- **Happy-path coverage.** The core scenario each AC describes\n  actually works in the code.\n- **Test-AC linkage.** Each AC should trace to at least one assert,\n  ideally with a test name that references the AC\'s verb.\n- **Verification command green.** If the card\'s Verification command\n  fails on the diff, the AC isn\'t truly satisfied even if the code\n  "looks right."\n- **No AC silently dropped.** If the diff deliberately doesn\'t\n  address an AC, the PR description must acknowledge it (and it\n  should probably be a REVISE for re-scoping).\n\n## What NOT to flag\n\n- Code quality on correct code (`code_quality`).\n- Robustness beyond what the AC requires (`robustness`).\n- Security concerns unrelated to the ACs (`security_code`).\n- Plan deviation (that\'s `plan_fidelity`).\n\n## Scoring\n\n- **10**: Every AC is ADDRESSED. Tests (or existing-test citations)\n  map 1:1 to ACs. Verification command is green.\n- **9**: Every AC ADDRESSED; one test could be more specific.\n- **8**: All ACs at least PARTIALLY, one with a minor gap an agent\n  could close at PR time.\n- **7**: One AC UNADDRESSED OR one AC PARTIALLY with a non-trivial\n  gap.\n- **≤6**: Multiple ACs UNADDRESSED / PARTIALLY, or the diff is\n  fundamentally incomplete against the card.\n\n## REVISE criteria\n\nREVISE when you can name:\n  (a) the AC number + text,\n  (b) what\'s missing in the diff,\n  (c) what needs to be added — a specific test, a handling branch,\n      an error path.\n\n## BLOCK only for\n\n- Verification command fails on the diff with no plausible path\n  to green.\n- Half or more of the ACs are UNADDRESSED.\n\n## CARD: escalation\n\nIf an AC is genuinely unimplementable given the non-goals + files\nscope of the card, use `CARD:` prefix. The author either needs to\nrelax the non-goals, broaden `files_expected`, or soften the AC.\nDon\'t grind the implementer on an impossible constraint.\n\n\n## When the critique is about the CARD, not the code\n\nRare at PR-phase — most card-shape issues were caught at plan-phase.\nBut if a card-level inconsistency forces the code to be wrong (AC\ndemands behavior non-goals forbid, Verification command targets files\nthe diff can\'t produce, etc.), use the **`CARD:`** prefix on your\ncomment. The aggregator routes CARD:+BLOCK verdicts to\n`QUESTION_FOR_AUTHOR` so the card owner can resolve, not the\nimplementer.\n\nExamples:\n- ✅ `CARD: AC 3 demands accepting raw HTML input; non-goals forbid\n  sanitization; the diff ships an XSS surface. Author needs to\n  reconcile.`\n- ❌ `CARD: the code should use parameterized queries` ← code-level\n  critique, not card-level. Drop the CARD: prefix.\n\n## How to choose a verdict\n\n**PROCEED** when:\n- Every AC is ADDRESSED in Step 1.\n- No non-goal is violated.\n- Your rubric lens shows no code-level concern that rises to "this PR\n  shouldn\'t merge as-is".\n- Nits and minor suggestions can go in `comments` — they\'re\n  information, not blockers.\n\n**Calibrate your scoring carefully.** A diff with no concerns in your\ndomain is a 9-10 depending on whether the code actively demonstrates\ndefenses you can name. **A pure-data diff that touches nothing your\nrubric cares about should score 8-9 with a brief "no surface" note —\nNOT 9.8+.** Inflation of the mean on orthogonal diffs is the major\npathology we\'re correcting for; keep the top of the scale reserved\nfor actively-defended work.\n\n**REVISE** when:\n- One or more AC items are UNADDRESSED / PARTIALLY. Cite the AC\n  number and say what\'s missing.\n- A non-goal is violated. Cite the non-goal and the diff line.\n- Your rubric found a concrete code-level issue the agent should fix\n  before merge — name file + line + the fix.\n\nREVISE comments are CITATIONS + CONCRETE FIXES, not vague\nskepticism. "This could be cleaner" is PROCEED-with-comments.\n\n**BLOCK** for:\n- Code is fundamentally wrong from your lens: a hardcoded secret in\n  the diff, a data-loss migration with no rollback, an\n  unauthenticated endpoint to sensitive data. Something no PR-round\n  revision will fix without rearchitecting.\n- The verification command fails and the diff can\'t plausibly be\n  iterated to green.\n\nBLOCK is heavy — use sparingly. The aggregator interprets BLOCK as\n"halt the loop and escalate" (or, if CARD:-prefixed, route to the\nauthor).\n\n## Respect the division of labor\n\nYou are ONE reviewer. Other reviewers cover other lenses:\n- `code_quality` — craftsmanship (readability, placement, idiom)\n- `security_code` — exploits (injection, secrets, authz)\n- `robustness` — runtime behavior (edge cases, failure paths)\n- `plan_fidelity` — does the diff implement the approved plan\n- `ac_verification` — does the diff satisfy each AC\n- `test_presence` — tests adequate for ACs\n- Language reviewers (`python_types`, `dart_flutter`) — language\n  idioms + type safety\n- Policy reviewers — diff vs linked ADRs / standards / policies\n\nDon\'t grade the whole PR on your single axis — focus on what your\nrubric actually covers. Don\'t score low because the PR has a concern\nthat belongs to another reviewer.\n\n## Score calibration\n\n- **10**: Exemplary in your domain. Code actively demonstrates the\n  defenses / properties your rubric cares about.\n- **9**: Strong. No concerns in your domain, or one minor nit that\'s\n  comment-worthy not REVISE-worthy.\n- **8**: Safe / sound in your domain. Maybe the PR has nothing your\n  rubric materially cares about — a brief "no surface" note and an\n  8 is correct (avoid 9.8 score-inflation).\n- **7**: Works but has a specific concern in your domain that\n  should be tightened before merge.\n- **6 or below**: Concrete problem that makes this PR not\n  merge-ready from your lens.\n\n## Evidence rules — every REVISE claim must be backed by a citation\n\nBefore you claim anything about the code or tests, locate the specific\n`file:line` in the diff. **Claims without citations are invalid — the\naggregator will treat them as noise.** Example of a good claim:\n`lib/foo.dart:42 — missing null check before dereference`. Example\nof a bad claim: `the error handling is weak`.\n\n**Verify before flagging.** The most common failure mode of PR review\n(observed on smoke9 mimir#143 R2/R3, where three reviewers each\nhallucinated that test assertions were missing) is stopping after the\nfirst thing you see and concluding without reading the full context.\nBefore you say "the tests don\'t assert X":\n\n1. **Read the full assertion chain, not just the first line.** In\n   Dart, `throwsA(isA<E>().having(...).having(...).having(...))`\n   CHAINS multiple `.having(...)` calls on a single expectation. Look\n   at EVERY `.having` call before concluding what the test asserts —\n   don\'t stop at the first one you see.\n   In Python, `pytest.raises(..., match=r"...")` takes a regex —\n   read the full pattern. In JS, `expect(...).toThrow(...)` can\n   accept strings, regexes, or matcher objects — check which.\n2. **Grep the diff for X.** Before claiming "the diff doesn\'t have\n   X", search the `diff_text` for X. If X appears, you cannot make\n   that claim. Either rewrite your finding to describe a different\n   gap (with citation) or mark this axis PROCEED on that point.\n3. **Trust the diff over your intuition.** If the diff shows the\n   behavior IS covered and your instinct says it should have been\n   flagged, your instinct is wrong. Code review is reading the code\n   that\'s there, not inventing what you thought should be there.\n\n## JSON output contract\n\nReturn ONLY a JSON object. REVISE and BLOCK verdicts MUST include at\nleast one entry in `findings` with a concrete `file:line` citation\nfrom the diff. PROCEED may have an empty `findings` array.\n\n{\n  "score": <float 0-10>,\n  "result": "PROCEED" | "REVISE" | "BLOCK",\n  "ac_coverage": {"1": "ADDRESSED" | "UNADDRESSED" | "PARTIALLY", "2": "...", ...},\n  "comments": "one paragraph of concrete feedback about the DIFF",\n  "findings": [\n    {\n      "file":         "lib/path/to/file.ext",\n      "line":         <int or null: the post-change line number from\n                      the diff\'s `+` markers. Use null if the gap is\n                      about something NOT present in the diff>,\n      "severity":     "blocker" | "major" | "minor",\n      "description":  "one-sentence specific gap — what\'s wrong",\n      "evidence":     "1-3 quoted lines FROM the diff text OR\n                      \'absent: <what should be there>\' if the gap\n                      is about missing content"\n    }\n  ]\n}\n\n**Rules for findings:**\n- A REVISE verdict with an empty `findings` array is malformed. The\n  aggregator will coerce such a verdict to PROCEED because the\n  reviewer couldn\'t back up the claim with evidence.\n- `evidence` MUST quote actual diff lines or explicitly say\n  `absent: <description>`. If you can\'t quote a line AND can\'t\n  articulate what\'s absent, the concern probably doesn\'t exist —\n  drop it.\n- `line` refers to the post-change line number (the one after `+`).\n  If the gap is about missing content (nothing was added), use\n  `null` and put `absent: ...` in `evidence`.\n- `severity`: `blocker` for real bugs / security / correctness\n  violations; `major` for plan-fidelity / AC-gap / design issues;\n  `minor` for style / naming / nits.\n\nNo prose outside the JSON.\n]\n\nCard title: [BENCH t01 deepseek-v3.2-c] feat(tags): Add common_tags() helper for standardized resource tagging\nEvaluate the PR diff (shown in your system prompt) against the card + approved plan + any typed policy links.  Return the JSON verdict per the contract.', '-Q', '--yolo', '--max-turns', '5']' timed out after 300 seconds]

</details>

<details>
<summary>✅ <strong>code_quality</strong> — 8.80/10</summary>

Clean, proportionate change for a small helper. `common_tags()` is placed in a dedicated module, uses straightforward normalization and validation, and returns the required dict without over-engineering. Naming is clear (`normalized_env`, `normalized_team`, `normalized_project`), and the implementation stays dependency-free and scoped to the two expected files. The tests are similarly direct and readable. Minor craftsmanship nits only: the `# type: ignore` on the pytest import looks unnecessary in this diff, and adding a blank line between test functions would better match typical pytest/Python formatting conventions, but neither issue rises to revise-before-merge.

</details>

<details>
<summary>✅ <strong>plan_fidelity</strong> — 7.60/10</summary>

The code change is tightly on-plan: it creates only the two approved files, implements `common_tags()` with the exact normalization/validation/return-shape the plan described, and adds the four planned pytest cases. The only plan-fidelity gap I can audit from the material provided is step 6 / AC 2: the diff shows the tests were added, but there is no evidence in the PR artifact here that the required repository-root `pytest` verification was actually run successfully after the change.

</details>

<details>
<summary>✅ <strong>robustness</strong> — 8.80/10</summary>

The runtime surface here is small and the behavior is deterministic on the inputs the card defines: all three parameters are normalized before use, invalid environments fail loudly with a ValueError instead of producing partial tags, and the returned mapping is fixed and side-effect free. I do not see a robustness issue that should block merge. Minor note only: the ValueError message interpolates a Python set, so the displayed order of allowed environments is not guaranteed, which can make diagnostics slightly less stable, but it does not affect correctness.

</details>

<details>
<summary>✅ <strong>security_code</strong> — 8.40/10</summary>

From a security-code perspective, this diff is low risk and merge-ready: infiquetra_aws_infra/tagging.py adds a small pure helper that only strips strings, validates env against a fixed allowlist, and returns a constant-shaped dict, so it does not introduce secrets exposure, injection sinks, auth/authz surface, or external-call/resource-exhaustion risk. The tests in tests/test_tagging.py likewise add no meaningful security surface. I marked AC2 as PARTIALLY only because the diff shows the tests were added, but the provided PR evidence does not include the actual pytest execution result.

</details>

<details>
<summary>✅ <strong>python_types</strong> — 9.00/10</summary>

From the Python typing lens, this diff is merge-ready. The new public API is fully annotated with precise parameter and return types in infiquetra_aws_infra/tagging.py, and the implementation matches the declared type: it always returns a dict[str, str] on success and raises on invalid env values. The tests are also annotated with -> None and import/use the helper consistently. I do not see any incorrect annotations, bare container types, Optional mismatches, or unnecessary Any/cast usage in the added code.

</details>

<details>
<summary>✅ <strong>test_presence</strong> — 9.50/10</summary>

The diff adds the exact test file the card required and the four test functions map cleanly to the observable behaviors in the task: valid env handling, invalid env raising, whitespace normalization, and required key presence including ManagedBy=CDK. Assertions are meaningful rather than no-op, and the new test path under tests/ matches the repo’s stated pytest collection configuration and the verification command from the card.

</details>

**Comment by namredips** [comment]
✅ Benchmark experiment reached merge-ready consensus. Closing without merging to keep `main` pristine for subsequent matrix runs. Scores live in the orchestrator SQLite `pr_reviews` table; artifacts are preserved in the branch and this PR history.
